src/pages/index.js

```diff
@@ -20,327 +24,243 @@ import UserInfo from "../components/UserInfo.js";
+  },
+});
+//***USER INFORMATION FROM THE API***
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo({
+      name: res.name,
+      description: res.about,
+    });
+    userInfo.setUserAvatar(res.avatar);
+  })
+  .catch(console.error);
 
-//***NEW SECTION***
 const cardsContainer = new Section(
-  { items: initialCards, renderer: renderCard },
+  { items: [], renderer: renderCard },
   ".cards__list"
 );
+//*** INITIAL CARDS FROM THE API***
+api
+  .getInitialCards()
+  .then((res) => {
+    //***RENDERER METHOD ON CARDSCONTAINER***
 
-//***RENDER CARDSCONTAINER***
-cardsContainer.rendererItems();
+    cardsContainer.rendererItems(res);
+  })
+  .catch(console.error);
+
+//***PREVIEW IMAGE***
+const previewPopup = new PopupWithImage("#image-preview-modal");
 
 //***POPUPWITHFORM***
-const profileEditModal = new PopupWithForm(
+const profileEditPopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileFormSubmit,
   config
 );
+/*/***FUNCTION HANDLING DELETE SUBMIT***
+function handleDeleteFormSubmit(values) {
+  deleteConfirmationPopup.renderSaving(true);
+  api
+    .deleteCard(values)
+    .then((res) => {
+      const deleteCard = renderCard(res);
+      cardsContainer.deleteItem(deleteCard);
+      formValidators.deleteForm.resetForm();
+      deleteConfirmationPopup.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      deleteConfirmationPopup.renderSaving(false);
+    });
+}
+
+*/
+
+//***FUNCTION HANDLING DELETE CLICK***
+function handleDeleteClick(card) {
+  //console.log(card);
+  deleteConfirmationPopup.open();
+  deleteConfirmationPopup.setCallback(() => {
+    deleteConfirmationPopup.renderSaving(true);
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.deleteCard();
+        deleteConfirmationPopup.close();
+      })
+      .catch(console.error)
+      .finally(() => {
+        deleteConfirmationPopup.renderSaving(false);
+      });
+  });
+}
+
+//***FUNCTION HANDLING LIKE CLICK***
+function handleLikeClick(card) {
+  api
+    .likeCard(card.getId(), card.isLiked)
+    .then((res) => card.toggleLikeCard(res.isLiked))
+    .catch(console.error);
+}
+
+//***FUNCTION HANDLING AVATAR EDIT SUBMIT***
+function handleAvatarFormSubmit(values) {
+  avatarEditPopup.renderSaving(true);
+  api
+    .updateAvatar(values)
+    .then((res) => {
+      userInfo.setUserAvatar(res.avatar);
+      avatarEditPopup.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      avatarEditPopup.renderSaving(false);
+    });
+}
 
+//***FUNCTION HANDLING PROFILE EDIT SUBMIT***
+function handleProfileFormSubmit(values) {
+  profileEditPopup.renderSaving(true);
+  api
+    .editProfile(values)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        description: res.about,
+      });
+      profileEditPopup.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      profileEditPopup.renderSaving(false);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```